lib/wallet.js

```diff
@@ -124,6 +130,156 @@ class Wallet extends ethers.Wallet {
+        
+        const exchangeContract = _exchangeContract.get(this);
+        return await exchangeContract.settleDriipAsPayment(receipt.toJSON(), this.address, options);
+    }
+
+    /**
+     * Without an amount of ETH from nahmii to base layer.
+     * @param {string} amountEth - The amount of ETH to withdraw from nahmii.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.withdrawEth('1.1', {gasLimit: 200000});
+     */
+    async withdrawEth(amountEth, options = {}) {
+        const currency = '0x0000000000000000000000000000000000000000';
+        const amountWei = ethers.utils.parseEther(amountEth.toString());
+        const clientFund = _clientFund.get(this);
+        return await clientFund.withdraw(amountWei, currency, 0, '', options);
+    }
+
+    /**
+     * Without an amount of ETH from nahmii to base layer.
+     * @param {string} amount - The amount of ETH to withdraw from nahmii.
+     * @param {string} symbol - The token symbol to withdraw.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.withdrawToken('1.1', 'TT1', {gasLimit: 200000});
+     */
+    async withdrawToken(amount, symbol, options = {}) {
+        const tokenInfo = await getTokenInfo.call(this, symbol);
+        const amountBN = ethers.utils.parseUnits(amount.toString(), tokenInfo.decimals);
+
+        const clientFund = _clientFund.get(this);
+        return await clientFund.withdraw(amountBN, tokenInfo.currency, 0, '', options);
+    }
+
+    /**
+     * Unstage an amount of ETH from staged balance back to nahmii available balance.
+     * @param {string} amount - The amount of ETH to unstage from staged balance.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.unstageEth('1.1', {gasLimit: 200000});
+     */
+    async unstageEth(amount, options = {}) {
+        const currency = '0x0000000000000000000000000000000000000000';
+        const amountWei = ethers.utils.parseEther(amount.toString());
+        const clientFund = _clientFund.get(this);
+        return await clientFund.unstage(amountWei, currency, 0, options);
+    }
+
+    /**
+     * Unstage an amount of ERC20 token from staged balance back to nahmii available balance.
+     * @param {string} amount - The amount of ERC20 token from staged balance to unstage balance(available balance).
+     * @param {string} symbol - The token symbol to unstage.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.unstageToken('1.1', 'TT1', {gasLimit: 200000});
+     */
+    async unstageToken(amount, symbol, options = {}) {
```

Comment:
Maybe this could be expanded to handle ETH also, so you don't need two sets of functions? Wouldn't it be better to have one function that could handle both ETH and tokens?

Reply:
Yes, that would be better. Let me consolidate the functions into one.